extensions-core/lookups-cached-global/src/main/java/io/druid/server/lookup/namespace/cache/UpdateCounter.java

```diff
@@ -34,14 +39,39 @@ void update()
 
   void awaitTotalUpdates(int totalUpdates) throws InterruptedException
   {
+    totalUpdates &= MAX_PHASE;
     int currentUpdates = phaser.getPhase();
-    while (totalUpdates - currentUpdates > 0) { // overflow-aware
+    checkNotTerminated(currentUpdates);
+    while (comparePhases(totalUpdates, currentUpdates) > 0) {
       currentUpdates = phaser.awaitAdvanceInterruptibly(currentUpdates);
+      checkNotTerminated(currentUpdates);
+    }
+  }
+
+  private static int comparePhases(int phase1, int phase2)
+  {
+    int diff = (phase1 - phase2) & MAX_PHASE;
```

Comment:
Let me suppose phase1 = Integer.MIN_VALUE and phase2 = 0. The diff value will also be 0 in this case. Is it a valid behavior?